src/app/core/modals/publish-modal/publish-modal.component.ts

```diff
@@ -128,22 +128,22 @@ export class PublishModalComponent extends DirectiveBase implements OnInit {
             })));
     }
 
-    onSubmit(): Promise<string> {
+    onSubmit(): Promise<void | { app: object, id: string}> {
         const {revisionNote, appID, content} = this.outputForm.getRawValue();
 
         this.isPublishing = true;
-        let saveCall: Promise<any>;
+        let saveCall: Promise<string>;
 
         if (this.revision === 0) {
             saveCall = this.platformRepository.createApp(appID, content);
         } else {
             saveCall = this.platformRepository.saveAppRevision(appID, content, revisionNote);
         }
 
-        return saveCall.then(() => {
+        return saveCall.then((str) => {
             this.isPublishing = false;
             this.close();
-            return appID;
+            return { app: JSON.parse(str), id: appID};
```

Comment:
Trying to push to the platform throws an error here

>core.js:1350 ERROR Error: Uncaught (in promise): SyntaxError: Unexpected token o in JSON at position 1
SyntaxError: Unexpected token o in JSON at position 1

Reply:
Apps should always be parsed as YAML before sending (which might also break, so try/catch in any case).